contracts/extensions/ExpenditureUtils.sol

```diff
@@ -0,0 +1,222 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./../colony/ColonyDataTypes.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./ColonyExtensionMeta.sol";
+
+// ignore-file-swc-108
+
+
+contract ExpenditureUtils is ColonyExtensionMeta, PatriciaTreeProofs {
+
+  // Storage
+
+  uint256 stakeFraction;
+
+  mapping (address => mapping (uint256 => uint256)) stakes;
```

Comment:
I guess it's nice that this way we're saving setting a storage slot for the creator, but this mapping doesn't feel like it's hugely useful? A mapping of `expenditureId => struct { creator, stake }`, or similar, feels like it'd be the alternative?

Getting the creator isn't _too_ bad from the frontend's perspective (find the first `ExpenditureTransferred` event for this expenditure, and see who it was transferred to), so I think I could go either way.

Reply:
Fair enough, I see how tracking down the creator could be tedious.